src/server.rs

```diff
@@ -1,89 +1,138 @@
+use std::{ffi::c_void, ptr};
+
+use open62541_sys::{UA_Server_runUntilInterrupt, __UA_Server_addNode, __UA_Server_write};
+
 use crate::{ua, DataType, Error, ObjectNode, Result, VariableNode};
 
+/// OPC UA server.
+///
+/// This represents an OPC UA server. Nodes can be added through the several methods below, and then
+/// the server can be started with [`run()`](Self::run).
 pub struct Server(ua::Server);
 
 impl Server {
-    /// Creates a new OPC UA server.
-    ///
-    /// # Errors
-    ///
-    /// When the OPC-UA server fails to initialize.
-    ///
-    pub fn new() -> Result<Self> {
-        Ok(Self(ua::Server::new()))
+    /// Creates server.
+    #[must_use]
+    pub fn new() -> Self {
```

Comment:
Why didn't clippy catch the needless `Result` that should be covered by `unnecessary_wraps`?